xpk.py

```diff
@@ -91,6 +91,15 @@
             spec:
               schedulerName: {args.scheduler}
               restartPolicy: Never
+              affinity:
+                nodeAffinity:
+                  requiredDuringSchedulingIgnoredDuringExecution:
+                    nodeSelectorTerms:
+                    - matchExpressions:
+                      - key: cloud.google.com/gke-nodepool
+                        operator: NotIn
+                        values:
+                        - default-pool
```

Comment:
What do you think of setting this only for CPUs?

Also would appreciate a comment saying that this "The scheduler can't schedule the Pod in the default-pool.".

Reply:
I left it here because it is a no-op for TPUs and GPUs, considering that we don't have the workload pods run on CPUs of default pool anyway. I could move this to be CPU-specific though and will add a comment (thanks for noticing that.)